compiler/rustc_next_trait_solver/src/solve/assembly/mod.rs

```diff
@@ -284,6 +284,14 @@ where
     ) -> Vec<Candidate<I>>;
 }
 
+pub(super) enum AssembleCandidatesFrom {
+    All,
+    /// Only assemble candidates from the environment and alias bounds, ignoring
+    /// /user-written and built-in impls. We only expect `ParamEnv` and `AliasBound`
```

Comment:
/user-written

also, pls add a quick comment to the enum itself describing why its necessary